pallets/settlement/src/lib.rs

```diff
@@ -919,9 +919,11 @@ impl<T: Trait> Module<T> {
             &[AffirmationStatus::Affirmed],
         )?;
         // Unlock tokens that were previously locked during the affirmation
-        let legs = <InstructionLegs<T>>::iter_prefix(instruction_id);
-        for (leg_id, leg_details) in
-            legs.filter(|(_leg_id, leg_details)| portfolios.contains(&leg_details.from))
+        let legs = <InstructionLegs<T>>::iter_prefix(instruction_id).collect::<Vec<_>>();
+        let legs_count = legs.len() as u32;
+        for (leg_id, leg_details) in legs
+            .into_iter()
+            .filter(|(_leg_id, leg_details)| portfolios.contains(&leg_details.from))
```

Comment:
Since this is repeated 3x, could we extract a function `-> (u32, impl Iterator<Item = Leg<?>>)` for this purpose?